src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -1461,6 +1536,135 @@ SliceLastMatmulMultiply::SliceLastMatmulMultiply() {
+        auto matched_node_transpose_in = node_to_output.at(transpose_in).get_node_shared_ptr();
+        auto matched_node_transpose_out = node_to_output.at(transpose_out).get_node_shared_ptr();
+        auto matched_node_multiply = node_to_output.at(multiply).get_node_shared_ptr();
+        auto matched_node_conv = node_to_output.at(conv).get_node_shared_ptr();
+
+        auto matched_param = std::static_pointer_cast<ov::op::v0::Parameter>(matched_node_param);
+        auto matched_param2 = std::static_pointer_cast<ov::op::v0::Parameter>(matched_node_param2);
+
+        const auto& shape = matched_param->get_shape();
+        const auto& shape2 = matched_param2->get_shape();
+        const auto& tr_in_shape = matched_node_transpose_in->input(0).get_shape();
+        const auto& tr_out_shape = matched_node_transpose_out->output(0).get_shape();
+
+        if (matched_param->get_element_type() == ov::element::i4 &&
+            matched_param2->get_element_type() == ov::element::f32 && shape.size() == 4 && shape2.size() == 4 &&
+            shape[2] == 1 && shape[3] == 1 && shape2[2] == 1 && shape2[3] == 1 && tr_in_shape.size() == 4 &&
+            tr_out_shape.size() == 4 && tr_in_shape[0] == 1 && tr_in_shape[1] == 1 && tr_out_shape[0] == 1 &&
+            tr_out_shape[1] == 1) {
+            auto new_param = std::make_shared<ov::op::v0::Parameter>(matched_param->get_element_type(),
+                                                                     ov::Shape{shape[0], shape[1]});
+            auto new_param2 = std::make_shared<ov::op::v0::Parameter>(matched_param2->get_element_type(),
+                                                                      ov::Shape{shape2[0], shape2[1]});
+
+            ctx.get().closures_to_reshape_remove.insert(matched_param);
+            ctx.get().closures_to_reshape_remove.insert(matched_param2);
+            ctx.get().closures_to_reshape_add.insert(new_param);
+            ctx.get().closures_to_reshape_add.insert(new_param2);
+
+            matched_node_convert->input(0).replace_source_output(new_param);
+            matched_node_convert->validate_and_infer_types();
+            matched_node_multiply->input(1).replace_source_output(new_param2);
+            matched_node_multiply->validate_and_infer_types();
+
+            auto matmul =
+                std::make_shared<ov::op::v0::MatMul>(matched_node_tr_input, matched_node_multiply, false, true);
+
+            for (auto&& r : matched_node_transpose_out->output(0).get_target_inputs()) {
+                r.replace_source_output(matmul);
+            }
+            return true;  // root has changed
+        }
+        return false;  // root hasn't changed
+    };
+    register_matcher(std::make_shared<opp::Matcher>(transpose_out, "ConvToMatmul"), std::move(callback));
+}
+
+ConvToMatmul2::ConvToMatmul2(Context::Ref ctx) {
+    auto param = opp::wrap_type<ov::op::v0::Parameter>();
```

Comment:
All my comments to the above pass apply to this one as well.

Reply:
Combined into a single pass